### PR TITLE
feat: add snapshot

### PR DESCRIPTION
**Pull request by FrankYang0529**
ref: https://github.com/harvester/harvester/issues/553

### CODE REVIEW

File: docs/vm/backup-restore.md [hunk 1]
```diff
@@ -116,6 +116,50 @@ EOF
...
+## Create a VM snapshot
+
+1. Go to the `Virtual Machines` page.
+1. Click `Take VM Snapshot` of the VM actions to create a new VM snapshot.
+1. Set a custom snapshot name and click `Create` to create a new VM snapshot.
+![create-snapshot.png](/img/v1.1/vm/create-snapshot.png)
+
+**Result:** The snapshot is created. You can also go to the `Backup & Snapshot > VM Snapshots` page to view all VM snapshots.
+
+The `ReadyToUse` status will be set to `true` once the Snapshot is complete.
```
Comment (guangbochen) [inline docs/vm/backup-restore.md:139]:
```suggestion
The `ReadyToUse` status will be set to `true` once the snapshot is complete.
```

File: docs/vm/backup-restore.md [hunk 2]
```diff
@@ -116,6 +116,50 @@ EOF
...
+You can replace an existing VM using the snapshot.
+
+:::note
+
+You can only choose to retain the previous volumes.
+
+:::
+
+1. Go to the `VM Snapshots` page.
+1. Click `Create`.
```
Comment (guangbochen) [inline docs/vm/backup-restore.md:163]: didn't find the create option?
image

**Review by guangbochen — COMMENTED** [review]
Maybe it is better to organize the backup and snapshot into different groups on the right sidebar.
image

File: docs/vm/backup-restore.md [hunk 3]
```diff
@@ -116,6 +116,50 @@ EOF
...
+1. Go to the `Virtual Machines` page.
+1. Click `Take VM Snapshot` of the VM actions to create a new VM snapshot.
+1. Set a custom snapshot name and click `Create` to create a new VM snapshot.
+![create-snapshot.png](/img/v1.1/vm/create-snapshot.png)
+
+**Result:** The snapshot is created. You can also go to the `Backup & Snapshot > VM Snapshots` page to view all VM snapshots.
+
+The `ReadyToUse` status will be set to `true` once the Snapshot is complete.
+
+Users can either choose to restore a new VM or replace an existing VM using this snapshot.
```
Comment (bk201) [inline docs/vm/backup-restore.md:141]:
```suggestion
Users can either restore a new VM or replace an existing VM using this snapshot.
```

Comment (FrankYang0529) [reply docs/vm/backup-restore.md]: Updated it. Thank you.

File: docs/vm/backup-restore.md (hunk 2 above)
Comment (FrankYang0529) [reply docs/vm/backup-restore.md]: Updated it. Thank you.

File: docs/vm/backup-restore.md (hunk 1 above)
Comment (FrankYang0529) [reply docs/vm/backup-restore.md]: Updated it. Thank you.

**Review by futuretea — APPROVED** [review]
LGTM

**Review by vickyhella — APPROVED** [review]
LGTM and thanks!

**Review by bk201 — APPROVED** [review]

**Review by guangbochen — APPROVED** [review]
LGTM, thanks.

**Pull request merged**